src/main/java/codeu/controller/ProfilePagesServlet.java

```diff
@@ -38,6 +47,35 @@ void setUserStore(UserStore userStore) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-        request.getRequestDispatcher("/WEB-INF/view/profile-pages.jsp").forward(request, response);
+    String requestUrl = request.getRequestURI();
+    String username = requestUrl.substring("/users/".length());
```

Comment:
This is confusing at first, I'd do:

```java
private static final String USERNAME_INDEX = 7;
...
String username = requestUrl.substring(USERNAME_INDEX);
```

Then it's more clear what you are doing.